storage/generic.go

```diff
@@ -17,11 +17,14 @@
 package storage
 
 import (
+	"io"
+
 	"github.com/prometheus/prometheus/pkg/labels"
 )
 
 type genericQuerier interface {
 	LabelQuerier
+	io.Closer
```

Comment:
We use the explicit method in all of our other interfaces, we don't use io.Closer anywhere.

Reply:
Not in `LabelQuerier` anymore

Reply:
fair point

Reply:
This doesn't solve it, now you've defined a new interface which we use in some places but not others.

Reply:
This comment stands.

Reply:
ack

Reply:
fixed! Reverted and actually renamed interface to makes more (: